app/controllers/recipients_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipientsController < ApplicationController
+  
+  before_action :set_message
+
+  # GET /recipients/1
+  def show
+    if @message.authenticated?
+      @message.delete()
+    else
+      redirect_to action: 'authenticate'
+    end
+  end
+
+  # GET /recipients/1/authenticate
+  def authenticate
+  end
+
+  # POST /recipients/1
+  def authenticatedShow
```

Comment:
I see what you're doing here.  You're flagging the record if they provide a valid password first... which is fine.

> One of the things I'm uncertain about is what rails expects in terms of convention between endpoints and views. I would think that authentication support would require a distinct view. If the authentication form lives in its own view then does convention demand a corresponding endpoint / controller method?

In terms of controller methods and view conventions, here is a good reference: http://guides.rubyonrails.org/layouts_and_rendering.html

Basically, your `GET` methods with map to views with the same name.  In ruby, we use snake case method names and file names by default.  So your `authenticatedShow` would become `authenticate_show`, and even `authenticate` for brevity.  Then you would create a view called `authenticate.html.erb` which would render by default when hitting the method.  Make sense?